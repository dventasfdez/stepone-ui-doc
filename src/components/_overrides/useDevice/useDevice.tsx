import { useEffect, useState } from "react";

export interface IUseDevice {
  isMobile: boolean;
  isTablet: boolean;
  isDesktop: boolean;
}

/**
 *
 * @internal
 */
export const useDevice = (): IUseDevice => {
  const [isMobile, setIsMobile] = useState(typeof window !== "undefined" && window.innerWidth <= 672);
  const [isTablet, setIsTablet] = useState(typeof window !== "undefined" && window.innerWidth <= 1024);
  const [isDesktop, setIsDesktop] = useState((typeof window !== "undefined" && window.innerWidth > 1024) ?? true);

  const handleResize = () => {
    if (typeof window !== "undefined" && window.innerWidth <= 672) {
      setIsMobile(true);
      setIsTablet(false);
      setIsDesktop(false);
    } else if (typeof window !== "undefined" && window.innerWidth <= 1024) {
      setIsMobile(false);
      setIsTablet(true);
      setIsDesktop(false);
    } else {
      setIsMobile(false);
      setIsTablet(false);
      setIsDesktop(true);
    }
  };

  useEffect(() => {
    if (typeof window !== "undefined") {
      window.addEventListener("resize", handleResize);
      window.addEventListener("orientationchange", handleResize);
    }

    return () => {
      window.removeEventListener("resize", handleResize);
      window.removeEventListener("orientationchange", handleResize);
    };
  }, []);

  return { isMobile, isTablet, isDesktop };
};
