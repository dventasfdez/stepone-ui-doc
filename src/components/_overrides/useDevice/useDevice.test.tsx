import {render} from '@testing-library/react';
import React from 'react';
import {useDevice} from './useDevice';
const TestUseDevice = () => {
  const {isDesktop, isMobile, isTablet} = useDevice();
  return (
    <div>
      <div>isDesktop: {isDesktop ? 'true' : 'false'}</div>
      <div>isMobile: {isMobile ? 'true' : 'false'}</div>
      <div>isTablet: {isTablet ? 'true' : 'false'}</div>
    </div>
  );
};

describe('useDevice', () => {
  it('should be defined', () => {
    expect(useDevice).toBeDefined();
  });
  it('should return isDesktop, isMobile, isTablet', () => {
    const {container} = render(<TestUseDevice />);
    expect(container).toMatchSnapshot();
  });
  it('should return isDesktop, isMobile, isTablet', () => {
    Object.defineProperty(window, 'innerWidth', {writable: true, configurable: true, value: 1024});
    const {container} = render(<TestUseDevice />);
    expect(container).toMatchSnapshot();
    Object.defineProperty(window, 'innerWidth', {writable: true, configurable: true, value: 672});
    const {container: container2} = render(<TestUseDevice />);
    expect(container2).toMatchSnapshot();
    Object.defineProperty(window, 'innerWidth', {writable: true, configurable: true, value: 1025});
    const {container: container3} = render(<TestUseDevice />);
    expect(container3).toMatchSnapshot();
  });
});
