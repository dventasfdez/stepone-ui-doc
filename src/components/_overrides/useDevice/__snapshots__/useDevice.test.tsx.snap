// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useDevice should return isDesktop, isMobile, isTablet 1`] = `
<div>
  <div>
    <div>
      isDesktop: 
      false
    </div>
    <div>
      isMobile: 
      false
    </div>
    <div>
      isTablet: 
      true
    </div>
  </div>
</div>
`;

exports[`useDevice should return isDesktop, isMobile, isTablet 2`] = `
<div>
  <div>
    <div>
      isDesktop: 
      false
    </div>
    <div>
      isMobile: 
      false
    </div>
    <div>
      isTablet: 
      true
    </div>
  </div>
</div>
`;

exports[`useDevice should return isDesktop, isMobile, isTablet 3`] = `
<div>
  <div>
    <div>
      isDesktop: 
      false
    </div>
    <div>
      isMobile: 
      true
    </div>
    <div>
      isTablet: 
      true
    </div>
  </div>
</div>
`;

exports[`useDevice should return isDesktop, isMobile, isTablet 4`] = `
<div>
  <div>
    <div>
      isDesktop: 
      true
    </div>
    <div>
      isMobile: 
      false
    </div>
    <div>
      isTablet: 
      false
    </div>
  </div>
</div>
`;
